src/main/java/dude/command/CommandExecutionException.java

```diff
@@ -0,0 +1,18 @@
+package dude.command;
+
+/**
+ * An exception thrown when a command is syntactically correct but is invalid.
+ * Eg. done given to a task that is already done.
+ * Eg. deleting a index which does not exist.
+ */
+public class CommandExecutionException extends Exception {
```

Comment:
You may consider creating your own `DudeException` base class for your custom exceptions to extend from, so you can differentiate clearly between exceptions that you expect to happen in Dude, and other general unchecked exceptions.